fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/ArrayMatchAnyExpander.java

```diff
@@ -0,0 +1,125 @@
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.rules.rewrite.OneRewriteRuleFactory;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.expressions.functions.scalar.ArrayMatchAny;
+import org.apache.doris.nereids.trees.expressions.literal.Literal;
+import org.apache.doris.nereids.trees.expressions.literal.MapLiteral;
+import org.apache.doris.nereids.trees.expressions.literal.StringLiteral;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalFilter;
+import org.apache.doris.nereids.trees.plans.logical.LogicalOlapScan;
+import org.apache.doris.nereids.util.ExpressionUtils;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * ArrayMatchAnyExpander
+ *
+ *  expand array_match_any(col1, "a,b") to array_match_any(col1, "a,b", invertedIdxCtx)
+ *  if literal array , we should not expand just to go with no inverted index match
+ *   like array_match_any(["a", "b", "c", 'd'], "a,b") to array_match_any(col1, "a,b")
+ *
+ * after expand, we can evaluate it.
+ */
+public class ArrayMatchAnyExpander extends OneRewriteRuleFactory {
+    @Override
+    public Rule build() {
+        return logicalFilter(logicalOlapScan())
+                .when(filter -> ExpressionUtils.containsType(filter.getExpressions(), ArrayMatchAny.class))
+                .then(this::replace)
+                .toRule(RuleType.ARRAY_MATCH_ANY_EXPANDER);
+    }
+
+    Plan replace(LogicalFilter<LogicalOlapScan> filter) {
+        Set<Expression> sets = filter.getConjuncts().stream().map(expr -> {
+            if (expr instanceof ArrayMatchAny) {
+                return rewrite((ArrayMatchAny) expr);
+            } else {
+                return expr;
+            }
+        }).collect(ImmutableSet.toImmutableSet());
+        return filter.withConjuncts(sets);
+    }
+
+    private static Expression rewrite(ArrayMatchAny arrayMatchAny) {
+        if (arrayMatchAny.children().size() > 2) {
+            return arrayMatchAny;
+        }
+        // default inverted index ctx
+        String invertedIndexParser = InvertedIndexUtil.INVERTED_INDEX_PARSER_ENGLISH;
+        String invertedIndexParserMode = InvertedIndexUtil.INVERTED_INDEX_PARSER_COARSE_GRANULARITY;
+        List<Literal> keys = Lists.newArrayList(InvertedIndexUtil.INVERTED_INDEX_PARSER_KEY,
+                        InvertedIndexUtil.INVERTED_INDEX_PARSER_MODE_KEY).stream()
+                        .map(StringLiteral::new).collect(Collectors.toList());
+        List<Literal> values = Lists.newArrayList(invertedIndexParser, invertedIndexParserMode).stream()
+                .map(StringLiteral::new).collect(Collectors.toList());
```

Comment:
do not use stream api. when u need generate new Collection, please use newWithExpectedSize as far as possible